lib/tokenizer.js

```diff
@@ -295,19 +436,115 @@ class ByteOrderedTokenizer extends Tokenizer {
   }
 
   /**
-   * @param {Buffer|Array} value
+   * @param {Buffer} value
    * @returns {Buffer}
    */
   hash(value) {
-    return value;
+    // strip any trailing zeros as tokens with trailing zeros are equivalent
```

Comment:
I noticed the java driver does this.  I don't think it's required, but if it wasn't done here, would have to update compare/equals methods to account for it

Reply:
Makes sense